### PR TITLE
Make 'Debug options' a member of the top-level settings screen; minor refactoring.

### PR DESCRIPTION
**Pull request by twaddington**
**What's this do?**
This patch makes 'Debug options' a little easier to find by making it a member top-level settings screen (see screenshots). The item is hidden by default and tapping the "Build" item repeatedly enables the option as before. I've also done a bit of restyling and refactoring of the settings code to clean things up a bit.

**Why are we doing this? (w/ JIRA link if applicable)**
n/a

**How should this be tested? / Do these changes have associated tests?**
n/a

**Dependencies for merging? Releasing to production?**
n/a

**Has the application documentation been updated for these changes?**
n/a

**Did someone actually run this code to verify it works?**
@twaddington ran the Vanilla app.

**Screenshots**
- [SettingsFragmentMain](https://user-images.githubusercontent.com/107117/90707891-6ac6e400-e24d-11ea-98f9-3c4d5d2ec8e9.png)
- [SettingsFragmentDebug](https://user-images.githubusercontent.com/107117/90707886-68fd2080-e24d-11ea-894b-a0d61ee22f82.png)

### CODE REVIEW

File: simplified-ui-settings/src/main/java/org/nypl/simplified/ui/settings/SettingsFragmentDebug.kt [hunk 1]
```diff
@@ -45,42 +43,54 @@ import org.nypl.simplified.ui.toolbar.ToolbarHostType
 import org.slf4j.LoggerFactory
 
 /**
- * A fragment that shows the set of accounts in the current profile.
+ * A fragment that shows various debug options for testing app functionality at runtime.
```
Comment (io7m) [inline simplified-ui-settings/src/main/java/org/nypl/simplified/ui/settings/SettingsFragmentDebug.kt:46]: Oops. Embarassing that that stayed in here for this long...

**Review by io7m — APPROVED** [review]
Looks good!

File: simplified-ui-settings/src/main/res/xml/settings.xml [hunk 2]
```diff
@@ -1,46 +1,64 @@
...
 
-</androidx.preference.PreferenceScreen>
\ No newline at end of file
+    <Preference
+      android:icon="@drawable/ic_settings_build"
+      android:key="settingsBuild"
+      android:summary="@string/settingsBuildSummary"
+      android:title="@string/settingsBuild"
+      app:enableCopying="true"
+      tools:summary="b3069242e" />
```
Comment (io7m) [inline simplified-ui-settings/src/main/res/xml/settings.xml:54]: I'm disappointed that you didn't hide an easter egg in this hex constant.

File: simplified-ui-settings/src/main/java/org/nypl/simplified/ui/settings/SettingsFragmentDebug.kt (hunk 1 above)
Comment (twaddington) [reply simplified-ui-settings/src/main/java/org/nypl/simplified/ui/settings/SettingsFragmentDebug.kt:46]: Just a little copy pasta 🍝

File: simplified-ui-settings/src/main/res/xml/settings.xml (hunk 2 above)
Comment (twaddington) [reply simplified-ui-settings/src/main/res/xml/settings.xml:54]: Haha, I will next time!

**Comment by twaddington** [comment]
Just a rebase.

**Pull request merged**